File: hackerrank/easy/0004-permutation-equation/js/solution.mjs
// solution.mjs

export function permutationEquation(p) {
  const result = [];
  for (let x = 1; x <= p.length; x++) {
    const posOfX = p.indexOf(x) + 1;
    const y = p.indexOf(posOfX) + 1;
    result.push(y);
  }
  return result;
}

// --- demo run ---
console.log(permutationEquation([2, 3, 1]));       // [2, 3, 1]
console.log(permutationEquation([4, 3, 5, 1, 2])); // [1, 3, 5, 4, 2]

File: hackerrank/easy/0004-permutation-equation/python/solution.py
## solution.py

from typing import List

def permutation_equation(p: List[int]) -> List[int]:
    result = []
    for x in range(1, len(p) + 1):
        pos_of_x = p.index(x) + 1
        y = p.index(pos_of_x) + 1
        result.append(y)
    return result


# --- demo run ---
if __name__ == "__main__":
    print(permutation_equation([2, 3, 1]))       # [2, 3, 1]
    print(permutation_equation([4, 3, 5, 1, 2])) # [1, 3, 5, 4, 2]